src/app/loadout-drawer/loadout-apply.ts

```diff
@@ -930,14 +960,16 @@ function applyLoadoutMods(
       )
     );
 
+    const allModHashes = modHashes.concat(Object.values(modsByBucket).flat());
+
     const mods = modHashes.map((h) => defs.InventoryItem.get(h)).filter(isPluggableItem);
 
     // Early exit - if all the mods are already there, nothing to do
-    if (allModsAreAlreadyApplied(armor, modHashes)) {
+    if (allModsAreAlreadyApplied(armor, allModHashes)) {
       infoLog('loadout mods', 'all mods are already there. loadout already applied');
       setLoadoutState((state) => ({
         ...state,
-        modStates: modHashes.map((modHash) => ({ modHash, state: LoadoutModState.Applied })),
+        modStates: allModHashes.map((modHash) => ({ modHash, state: LoadoutModState.Applied })),
```

Comment:
For shaders, this isn't entirely correct because the condition is satisfied if the desired shader is already present on a different piece of armor.